### PR TITLE
feat: rework secureboot and PCR signing key

### PR DESCRIPTION
**Pull request by smira**
Support different providers, not only static file paths.

Drop `pcr-signing-key-public.pem` file, as we generate it on the fly now.

See https://github.com/siderolabs/image-factory/issues/19

### CODE REVIEW

**Comment by smira** [comment]
This PR should be mostly no-op for Talos itself, but it provides a way to generate assets signed with Azure Key Vault, which will be consumed in the Image Factory.

File: go.mod [hunk 1]
```diff
@@ -146,12 +149,20 @@ require (
 	sigs.k8s.io/yaml v1.4.0
 )
 
+require (
```
Comment (frezbo) [inline go.mod:152]: can we group this with others?

File: internal/pkg/secureboot/database/database.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	Name     string
+	Contents []byte
+}
+
+// Generate generates a UEFI database to enroll the signing certificate.
+//
+// ref: https://blog.hansenpartnership.com/the-meaning-of-all-the-uefi-keys/
+func Generate(enrolledCertificate []byte, signer pesign.CertificateSigner) ([]Entry, error) {
+	// derive UUID from enrolled certificate
+	uuid := uuid.NewHash(sha256.New(), uuid.NameSpaceX500, enrolledCertificate, 4)
```
Comment (frezbo) [inline internal/pkg/secureboot/database/database.go:31]: :cool:

File: pkg/imager/out.go [hunk 3]
```diff
@@ -94,7 +98,56 @@ func (i *Imager) outISO(path string, report *reporter.Reporter) error {
 
 			ScratchDir: scratchSpace,
 			OutPath:    path,
-		})
+		}
+
+		if i.prof.Input.SecureBoot.PlatformKeyPath == "" {
```
[the remark this answers is not in the document]
Comment (smira) [reply pkg/imager/out.go:103]: yes, if we were not given secure database keys explicitly, we generate them automatically on build

**Review by frezbo — APPROVED** [review]

**Pull request merged**